be/src/runtime/buffer_control_block.h

```diff
@@ -67,9 +67,12 @@ class BufferControlBlock {
     ~BufferControlBlock();
 
     Status init();
-    Status add_batch(TFetchDataResult* result);
+    Status add_batch(std::unique_ptr<TFetchDataResult>& result);
+    Status add_batch(std::vector<std::unique_ptr<TFetchDataResult>>& results);
+
     // non-blocking version of add_batch
-    StatusOr<bool> try_add_batch(TFetchDataResult* result);
+    StatusOr<bool> try_add_batch(std::unique_ptr<TFetchDataResult>& result);
```

Comment:
would better only change `try_add_batch` method for pipeline engine.  needn't change for non-pipeline engine.

Reply:
> would better only change `try_add_batch` method for pipeline engine. needn't change for non-pipeline engine.

I think the serialization issue exists in both pipeline and non-pipeline engine, if we don't change `add_batch`, it can't be avoided in non-pipeline engine.